src/ril_binder_radio.c

```diff
@@ -912,6 +939,141 @@ ril_binder_radio_encode_setup_data_call_1_2(
     return ok;
 }
 
+/**
+ * @param int32_t Serial number of request.
+ * @param RadioTechnology Radio technology to use.
+ * @param DataProfileInfo Data profile info.
+ * @param bool roamingAllowed Indicating data roaming is allowed or not.
+ */
+static
+gboolean
+ril_binder_radio_encode_setup_data_call_1_4(
+    GRilIoRequest* in,
+    GBinderLocalRequest* out)
+{
+    gboolean ok = FALSE;
+    GRilIoParser parser;
+    gint32 count, tech, auth, profile_id;
+    char* profile_str = NULL;
+    char* tech_str = NULL;
+    char* auth_str = NULL;
+    char* apn = NULL;
+    char* user = NULL;
+    char* password = NULL;
+    char* proto = NULL;
+
+    ril_binder_radio_init_parser(&parser, in);
+    if (grilio_parser_get_int32(&parser, &count) && count == 7 &&
+        (tech_str = grilio_parser_get_utf8(&parser)) != NULL &&
+        gutil_parse_int(tech_str, 10, &tech) &&
+        (profile_str = grilio_parser_get_utf8(&parser)) != NULL &&
+        gutil_parse_int(profile_str, 10, &profile_id) &&
+        (apn = grilio_parser_get_utf8(&parser)) != NULL &&
+        (user = grilio_parser_get_utf8(&parser)) != NULL &&
+        (password = grilio_parser_get_utf8(&parser)) != NULL &&
+        (auth_str = grilio_parser_get_utf8(&parser)) != NULL &&
+        gutil_parse_int(auth_str, 10, &auth) &&
+        (proto = grilio_parser_get_utf8(&parser)) != NULL) {
+        GBinderWriter writer;
+        RadioDataProfile_1_4* profile;
+        RADIO_ACCESS_NETWORK ran;
+
+        /* ril.h has this to say about the radio tech parameter:
+         *
+         * ((const char **)data)[0] Radio technology to use: 0-CDMA,
+         *                          1-GSM/UMTS, 2... for values above 2
+         *                          this is RIL_RadioTechnology + 2.
+         *
+         * Makes little sense but it is what it is.
+         */
+        if (tech > 4) {
+            tech -= 2;
+        }
+
+        ran = RADIO_ACCESS_NETWORK_UNKNOWN;
+        switch ((RADIO_TECH)tech) {
```

Comment:
Is this mapping different from the one in `ril_binder_radio_encode_setup_data_call_1_2()`? If not, please move it to a separate function and call it from both places.